### PR TITLE
tracer: use official otel spec for configuration

### PR DESCRIPTION
**Pull request by bobheadxi**
See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#configuration-options which I only just learned about today as part of #38889 - the original way we used `OTEL_EXPORTER_OTLP_ENDPOINT` was incorrect

## Test plan

https://docs.sourcegraph.com/dev/how-to/otel_local_dev

### CODE REVIEW

File: internal/tracer/otel.go [hunk 1]
```diff
@@ -69,26 +63,48 @@ func newOTelBridgeTracer(logger log.Logger, opts *options) (opentracing.Tracer,
...
+	// Set up client to otel-collector - we replicate some of the logic used internally in
+	// https://github.com/open-telemetry/opentelemetry-go/blob/21c1641831ca19e3acf341cc11459c87b9791f2f/exporters/otlp/internal/otlpconfig/envconfig.go
+	// based on our own inferred endpoint.
+	opts := []otlptracegrpc.Option{
+		otlptracegrpc.WithEndpoint(trimSchema(endpoint)),
 	}
+	if isInsecureEndpoint(endpoint) {
+		opts = append(opts, otlptracegrpc.WithInsecure())
+	}
+	client := otlptracegrpc.NewClient(opts...)
```
Comment (bobheadxi) [inline internal/tracer/otel.go:104]: this is how we should have originally used `OTEL_EXPORTER_OTLP_ENDPOINT`

Comment (jhchabran) [reply internal/tracer/otel.go:104]: How did you spot that our usage was incorrect? What were the symptoms?

File: internal/tracer/otel.go [hunk 2]
```diff
@@ -125,3 +141,13 @@ func newResource(r log.Resource) *resource.Resource {
...
 		semconv.ServiceVersionKey.String(r.Version))
 }
+
+func isInsecureEndpoint(endpoint string) bool {
+	return strings.HasPrefix(strings.ToLower(endpoint), "http://")
+}
+
+var httpSchemeRegexp = regexp.MustCompile(`(?i)^http://|https://`)
+
+func trimSchema(endpoint string) string {
```
Comment (jhchabran) [inline internal/tracer/otel.go:151]: nit: what about using `url.Parse` and picking the parts we want instead?

**Review by jhchabran — APPROVED** [review]

File: internal/tracer/otel.go (hunk 1 above)
Comment (bobheadxi) [reply internal/tracer/otel.go:104]: You can see the implementation here: https://sourcegraph.com/github.com/open-telemetry/opentelemetry-go/-/blob/exporters/otlp/otlptrace/internal/otlpconfig/envconfig.go?L58-68

**Pull request merged**